frame/society/src/lib.rs

```diff
@@ -470,8 +505,7 @@ impl<T: Trait> Module<T> {
 			let candidates = <Candidates<T>>::take();
 			members.reserve(candidates.len());
 
-			let maturity = <system::Module<T>>::block_number()
-				+ Self::lock_duration(members.len() as u32);
+			let maturity = Self::maturity();
```

Comment:
well done; you've just managed to double the time we spend decoding members.

Reply:
don't split out storage accesses into low-level getters without thinking.

Reply:
The takeaway for others here is:

> helpers shouldn't be pulling in storage; in general they shouldn't be using globals at all. it's forgivable if all conceivable instances of usage would all use those globals.

This pattern should be easy to identify upon review